### PR TITLE
Fix LogRecordTest by using InvariantCulture.

### LINKED ISSUE #1678: LogRecordTest.CheckStateForStrucutredLogWithGeneralType fails when run in locale with comma as decimal separator
Test failure:

```
 OpenTelemetry.Tests.Logs.LogRecordTest.CheckStateForStrucutredLogWithGeneralType
   Source: LogRecordTest.cs line 193
   Duration: 5 ms

  Message: 
    Assert.Equal() Failure
                                     ↓ (pos 28)
    Expected: ···ruffle], [Price, 299.99]
    Actual:   ···ruffle], [Price, 299,99]
                                     ↑ (pos 28)
  Stack Trace: 
    LogRecordTest.CheckStateForStrucutredLogWithGeneralType() line 216
```

It seems to fail in all frameworks.

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #1678.

## Changes

Enforce that the current Culture is InvariantCulture around the call to ToString on the log record state.

Note that using `String.Format(CultureInfo.InvariantCulture, "{0}", state)` does not work here.

### CODE REVIEW

**Review by eddynaka — APPROVED** [review]

**Comment by Oberon00** [comment]
Same as #1680: Can we merge this yet? I cannot give you access to my branch since it's an organization repository unfortunately.

**Review by cijothomas — APPROVED** [review]

File: test/OpenTelemetry.Tests/Logs/LogRecordTest.cs [hunk 1]
```diff
@@ -213,7 +214,16 @@ public void CheckStateForStrucutredLogWithGeneralType()
             Assert.Contains(state, item => item.Key == "{OriginalFormat}");
             Assert.Equal("{food}", state.First(item => item.Key == "{OriginalFormat}").Value);
 
-            Assert.Equal("[Name, truffle], [Price, 299.99]", state.ToString());
+            var prevCulture = CultureInfo.CurrentCulture;
```
Comment (reyang) [inline test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:217]: If there is a common need, consider make this a "using statement".

Comment (cijothomas) [reply test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:217]: also, we can move to shared extension method like https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/src/OpenTelemetry.Api/Internal/ExceptionExtensions.cs#L31

Comment (Oberon00) [reply test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:217]: This was the only test that failed that way, at least with a German locale.

**Pull request merged**